torchgeo/datasets/quakeset.py

```diff
@@ -0,0 +1,291 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""QuakeSet dataset."""
+
+import os
+from collections.abc import Callable
+from typing import Any, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import DatasetNotFoundError, download_url, percentile_normalization
+
+
+class QuakeSet(NonGeoDataset):
+    """QuakeSet dataset.
+
+    `QuakeSet <https://huggingface.co/datasets/DarthReca/quakeset>`__
+    is a dataset for Earthquake Change Detection and Magnitude Estimation and is used
+    for the Seismic Monitoring and Analysis (SMAC) ECML-PKDD 2024 Discovery Challenge.
+
+    Dataset features:
+
+    * Sentinel-1 SAR imagery
+    * before/pre/post imagery of areas affected by earthquakes
+    * 2 SAR bands (VV/VH)
+    * 3,327 pairs of pre and post images with 5 m per pixel resolution (512x512 px)
+    * 2 classification labels (unaffected / affected by earthquake)
+    * pre/post image pairs represent earthquake affected areas
+    * before/pre image pairs represent hard negative unaffected areas
+    * earthquake magnitudes for each sample
+
+    Dataset format:
+
+    * single hdf5 dataset containing images, magnitudes, hypercenters, and splits
+
+    Dataset classes:
+
+    0. unaffected area
+    1. earthquake affected area
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/2403.18116
+
+    .. note::
+
+       This dataset requires the following additional library to be installed:
+
+       * `h5py <https://pypi.org/project/h5py/>`_ to load the dataset
+
+    .. versionadded:: 0.6
+    """
+
+    filename = "earthquakes.h5"
+    url = "https://hf.co/datasets/DarthReca/quakeset/resolve/bead1d25fb9979dbf703f9ede3e8b349f73b29f7/earthquakes.h5"
+    md5 = "76fc7c76b7ca56f4844d852e175e1560"
+    splits = {"train": "train", "val": "validation", "test": "test"}
```

Comment:
Why not use "validation" instead of "val"? Then we don't need a dict to map.

Reply:
The dataset h5 file attributes uses "validation". All other torchgeo dataset splits use "val".

Reply:
> All other torchgeo dataset splits use "val".

This is by coincidence, not rule. So2Sat also uses "validation".

Reply:
Ahh okay will change